### PR TITLE
updateUsername & changes to createLibraryFromCollection

### PR DESCRIPTION
**Pull request by ahsu1230**
removed getLibraryByUserAndSlug from LibraryCommander

@andrewconner

### CODE REVIEW

**Comment by andrewconner** [comment]
Nice! Looks good.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala [hunk 1]
```diff
@@ -239,6 +239,22 @@ class UserController @Inject() (
...
 
+  def updateUsername() = JsonAction.authenticatedParseJson { implicit request =>
+    val newUsername = (request.body \ "username").as[Username]
+    userCommander.setUsername(request.userId, newUsername) match {
+      case Left(error) => BadRequest(Json.obj("error" -> error))
+      case Right(username) => Ok(Json.obj("username" -> username))
+    }
+  }
+
+  def updateName() = JsonAction.authenticatedParseJson { implicit request =>
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:250]: Mind making a route for this?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala [hunk 2]
```diff
@@ -239,6 +239,22 @@ class UserController @Inject() (
...
+    val newFirstName = (request.body \ "firstName").as[String]
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:251]: Mind making this and below an option?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala [hunk 3]
```diff
@@ -239,6 +239,22 @@ class UserController @Inject() (
...
+    val newLastName = (request.body \ "lastName").as[String]
+    val userData = UpdatableUserInfo(firstName = Some(newFirstName), lastName = Some(newLastName), description = None, emails = None)
+    userCommander.updateUserInfo(request.userId, userData)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:254]: We should move away from `UpdatableUserInfo`. It's vague what the caller wants to happen.

The commander should have a `updateName(userId: Id[User], firstName: Option[String], lastName: Option[String]` method.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala [hunk 4]
```diff
@@ -239,6 +239,22 @@ class UserController @Inject() (
...
+    Ok(JsString("success"))
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:255]: JSON actually is a subset of what's a possible js variable. Could this return an object?

File: kifi-backend/modules/shoebox/conf/shoebox.routes [hunk 5]
```diff
@@ -266,6 +266,7 @@ POST    /site/recos/vote            @com.keepit.controllers.website.Recommendati
 
 GET     /site/user/me               @com.keepit.controllers.website.UserController.currentUser()
 POST    /site/user/me               @com.keepit.controllers.website.UserController.updateCurrentUser()
+POST    /site/user/username         @com.keepit.controllers.website.UserController.updateUsername()
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/conf/shoebox.routes:269]: `/site/user/me/username` may make more sense.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala (hunk 4 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:255]: An object is certainly more flexible, but for the record, the JS string `"success"` is valid JSON, as are the non-string values `2` and `false`.

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:255]: what object do you think should be returned?
create a new User class that has first/last name, username, & email?

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:255]: I guess BasicUserInfo might do the trick

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserController.scala:255]: Interesting, I confused this with an empty response being invalid, apologies for the confusion. You're good.

**Pull request merged**